libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -810,6 +810,8 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 		!functionType->bound() &&
```

Comment:
This can just be removed. @cameel can you check, please?

Reply:
Yeah, it should be removed because you added a `m_context.internalFunctionAccessed()` call in `endVisit(MemberAccess const&)`. Otherwise it will unnecessarily add an entry to the internal dispatch.

Alternatively you could just leave this as is and not add the `m_context.internalFunctionAccessed()` call because they always cancel each other out (because pointers to bound functions are not allowed). Having them both is just unnecessary overhead. On the other hand it **is** an internal function access so leaving it in may make the code clearer.

BTW, you could add an assertion in `IRGenerationContext::internalFunctionAccessed()` to ensure that bound functions are not added to the dispatch. This would ensure that always either both of those calls are there or neither is and we don't get unnecessary entries.

Reply:
Turns out that removing `internalFunctionAccessed()` is not an option after all due to the `(x.f)()` case. Even more reason to add that assertion then.

Reply:
Which assertion do you mean?

Reply:
The one mentioned in the comment above. I said that an assertion in `IRGenerationContext::internalFunctionAccessed()` could check that the functions we're adding to the dispatch are never bound.

Reply:
I'm pretty sure this assertion will fail on `(x.f)()`, but I can check.

Reply:
Ah, you're right. No assertion needed. It won't work.